### ISSUE TITLE
Use `--build-arg` to specify versions during the build process

### ISSUE THREAD

**Issue by RA80533** [microsoft/codeql-container#25]
- https://github.com/microsoft/codeql-container/blob/d0e531a4b7db837e7e92e8d611b1226eb45ef861/Dockerfile#L58
- https://github.com/microsoft/codeql-container/blob/d0e531a4b7db837e7e92e8d611b1226eb45ef861/Dockerfile#L65-L66
- https://github.com/microsoft/codeql-container/blob/d0e531a4b7db837e7e92e8d611b1226eb45ef861/Dockerfile#L67-L68
- https://github.com/microsoft/codeql-container/blob/d0e531a4b7db837e7e92e8d611b1226eb45ef861/Dockerfile#L70-L73

These occurrences can be updated to vastly improve the idempotence of building the Docker image. Right now, running `docker build` produces different images based on when the command is run. It could change quite a bit [over the course of a week](https://github.com/github/codeql/pulse).

**Comment by jacobmsft** [comment]
Good point. I'll put this in the backlog, alternatively, I'll be happy to accept a pull request.
